library/azure_ad_users.py

```diff
@@ -0,0 +1,310 @@
+        except urllib2.HTTPError, err:
+            response_code = err.getcode()
+            response_msg = err.read()
+            response_json = json.loads(response_msg)
+            if response_json.get("odata.error", False) and "already exists" in response_json.get("odata.error").get("message",{}).get("value"):
+                self.module.exit_json(msg="User already exists", changed=False)
+            else:
+                error_msg = response_json.get("odata.error").get("message")
+                self.module.fail_json(msg="Error happend while trying to create user. Error code='{}' msg='{}'".format(response_code, error_msg))
+
+        self.module.exit_json(msg="User created.", changed=True)
+
+    def delete_user(self):
+        # https://msdn.microsoft.com/en-us/Library/Azure/Ad/Graph/api/users-operations#BasicoperationsonusersDeleteauser
+        #payload = { 'userPrincipalName': self.user_name }
+        #payload = json.dumps(payload)
+        url = self.graph_url + "/users/" + self.user_name + "?" + self.azure_version
+        #print (url)
+        #exit(1)
+        try:
+            r = open_url(url, method="delete", headers=self.headers) #,data=payload)
+        except urllib2.HTTPError, err:
+            response_code = err.getcode()
+            response_msg = err.read()
+            response_json = json.loads(response_msg)
+            if response_json.get("odata.error", False) and "Insufficient privileges" in response_json.get("odata.error").get("message",{}).get("value"):
+                self.module.exit_json(msg="You have to add this Service Principal to the \"User Account Administrator Role\" this can be done using Powershell.", changed=False)
+            else:
+                error_msg = response_json.get("odata.error").get("message")
+                self.module.fail_json(msg="Error happend while trying to delete the user. Error code='{}' msg='{}'".format(response_code, error_msg))
+
+        self.module.exit_json(msg="User deleted.", changed=True)
+
+    def main(self):
+        if self.state == "present":
+            if self.user_name.find('@')==1:
+                self.module.fail_json(msg="Please make sure to enter the username without the @tenant_domain.onmicrosoft.com")
+            if self.password == None:
+                self.module.fail_json(msg="You can't create a user without specifing a password!")
+            if self.display_name == None:
+                #i = self.user_name.split('@', 1)
+                self.display_name = self.user_name
+            self.login()
+            self.create_user()
+
+        elif self.state == "absent":
+            if self.user_name.find('@')==1:
+                self.module.fail_json(msg="Please make sure to enter the username without the @tenant_domain.onmicrosoft.com")
+            if self.password != None:
+                self.module.fail_json(msg="I am confiused, you specified a password for a user to be deleted! Are you sure its a delete operation?")
```

Comment:
I understand why you might do this. But I would vote against it. I would just delete even if password was given. And my idea is  if you have yaml/json definition for users. You might want to delete users who has a password set.